reframe/frontend/executors/__init__.py

```diff
@@ -168,14 +180,18 @@ def runall(self, checks, system):
             self._printer.timestamp('Started on', 'short double line')
             self._printer.info()
             self._runall(checks, system)
+            if self._max_retries > 0:
+                self._retry(checks, system)
+
         finally:
-            # Always update statistics and print the summary line
-            self._stats = self._policy.getstats()
-            num_failures = self._stats.num_failures()
+            # Print the summary line
+            num_failures_last_retry = self._stats.num_failures(try_num=-1)
```

Comment:
The `num_failures` should return the failures of the last retry by default.

Reply:
I had put `try_num=0` as default for all functions in stats for not to interfere with the calls from elsewhere in ReFrame as e.g. the unit tests in `TestSerialExecutionPolicy`. Should I make `try_num=-1` as default for all functions and assure that also in the rest of ReFrame it is called consistently?

Reply:
The most reasonable thing is to always report from the last run. What unit tests do, it should not affect the design. Also, I don't understand the "elsewhere in ReFrame." This method is only called in the `Runner` and in the failure report.

Reply:
OK, I made the last run to be the default for all functions. Done.